x/programs/runtime/runtime.go

```diff
@@ -36,12 +40,36 @@ type runtime struct {
 	engine   wazero.Runtime
 	mod      api.Module
 	meter    Meter
-	storage  Storage
 	// functions exported by this runtime
 	mu state.Mutable
 
 	closed bool
 	log    logging.Logger
+	// db for persistent storage
+	db database.Database
+	// runtime pubkey
+	callerAddress ed25519.PublicKey
+}
+
+func (r *runtime) initProgramStorage(programBytes []byte) (int64, error) {
+	count, err := GetProgramCount(r.db)
+	if err != nil {
+		r.log.Error("failed to get program counter", zap.Error(err))
+		return 0, err
+	}
+	// increment
+	err = IncrementProgramCount(r.db)
```

Comment:
These implementation details should not be part of the runtime IMO. This type of logic would be part of an action.

Reply:
this is future TODO